app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -316,6 +320,54 @@ protected function getRatesDebuggingData($rates) {
         return $rateDebuggingData;
     }
 
+    protected function isDiscountRoundingDeltaError($transaction, $quote) {
+        $boltDiscountAmount = $this->getBoltDiscountAmount($transaction);
+        $quoteDiscountAmount = $quote->getShippingAddress()->getDiscountAmount();
+
+        return $this->isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount);
+    }
+
+    protected function getBoltDiscountAmount($transaction) {
+        $boltDiscountAmount = 0;
+
+        if(isset($transaction->order->cart->discounts)) {
+            foreach($transaction->order->cart->discounts as $discount) {
+                if(isset($discount->amount->amount) && is_numeric($discount->amount->amount) && $discount->amount->amount > 0) {
+                    $boltDiscountAmount -= $discount->amount->amount / 100;
+                }
+            }
+        }
+
+        return $boltDiscountAmount;
+    }
+
+    protected function isOnePennyRoundingError($boltDiscountAmount, $quoteDiscountAmount) {
+        if($boltDiscountAmount != $quoteDiscountAmount) {
+            $difference = round(abs($boltDiscountAmount - $quoteDiscountAmount), 2);
+            if($difference <= 0.01) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    protected function fixQuoteDiscountAmount($transaction, $quote) {
+        $quote->setTotalsCollectedFlag(false)->collectTotals()->save();
+
+        if($this->isDiscountRoundingDeltaError($transaction, $quote)) {
```

Comment:
We should instead check `$boltDiscountAmount != $quoteDiscountAmount`

Reply:
That's essentially what this is doing, but it's checking to see if they are only off by $0.01. If they are off by more than this, then there is a different issue. Calling collectTotals could pull in other artifacts that could cause this number to be off by more than $0.01 (admin adds a new discount amount, etc) which would be a different error and would be caught by validateSubmittedOrder